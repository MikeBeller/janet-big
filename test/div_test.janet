(import ../build/big :as big)

(defn time-div [n d nreps]
  (def start (os/clock))
  (var x 0)
  (repeat nreps
          (set x (/ n d)))
  (def end (os/clock))
  (/ (- end start) nreps))

(each line (string/split "\n" (slurp "rands.txt"))
  (def f (string/split " " line))
  (def nbits (scan-number (in f 0)))
  (def dbits (scan-number (in f 1)))
  (def n (big/int (in f 2)))
  (def d (big/int (in f 3)))
  (def t (time-div n d 100))
  (def tns (math/floor (* t 1000000000)))
  (printf "%q %q %q" nbits dbits tns))


