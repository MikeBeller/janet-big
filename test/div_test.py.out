1000 100 669
1000 333 691
1000 500 662
1000 750 510
1000 999 197
10000 1000 16882
10000 3333 33614
10000 5000 40099
10000 7500 27799
10000 9999 1456
100000 10000 1207234
100000 33333 2882859
100000 50000 3222525
100000 75000 2415175
100000 99999 12311
300000 30000 10485548
300000 100000 25618846
300000 150000 28818230
300000 225000 21600735
300000 299999 33
